src/Microsoft.Health.Dicom.Core/Features/Store/StoreResponseBuilder.cs

```diff
@@ -85,6 +85,19 @@ public void AddSuccess(DicomDataset dicomDataset)
             referencedSopSequence.Items.Add(referencedSop);
         }
 
+        /// <inheritdoc/>
+        public void AddWarning(DicomDataset dicomDataset, ushort warningReasonCode)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+
+            CreateDatasetIfNeeded();
+
+            if (!dicomDataset.TryGetSequence(DicomTag.WarningReason, out DicomSequence _))
+            {
+                dicomDataset.Add(DicomTag.WarningReason, warningReasonCode);
```

Comment:
Looking at the way this should be returned according to https://dicom.nema.org/dicom/2013/output/chtml/part18/sect_6.6.html#table_6.6.1-2, the warning reason should be added to the Referenced SOP Sequence for each result. Since we are already populating that attribute in the AddSuccess method above, how about passing the warning reason as an optional parameter to AddSuccess and handling it there?

Reply:
Done